hw/ip/otbn/dv/otbnsim/sim/edn_client.py

```diff
@@ -61,13 +66,19 @@ def forget(self) -> None:
         '''
         self._retry = False
 
-    def take_word(self, word: int) -> None:
+    def take_word(self, word: int, fips_err: bool) -> None:
         '''Take a 32-bit data word that we've been waiting for'''
         assert 0 <= word < (1 << 32)
         assert self._acc is not None
         assert len(self._acc) < ACC_LEN
         assert self._cdc_counter is None
 
+        # If the length of accumulated words is nonzero we can check for
+        # repetition
+        if len(self._acc):
```

Comment:
Does this mean we don't check repetitions across 256-bit boundaries? Because the RTL also detects repetitions across this boundary. E.g. this RND input will still create an error in the RTL:
```
RND 0 [255:0]: 0x77777777 0x66666666 0x55555555 0x44444444 0x33333333 0x22222222 0x11111111 0x00000000
RND 1 [255:0]: 0xEEEEEEEE 0xDDDDDDDD 0xCCCCCCCC 0xBBBBBBBB 0xAAAAAAAA 0x99999999 0x88888888 0x77777777

```

Reply:
I thought this was about checking the repetition inside the received RND word. I checked the RTL about sending the error and I think you are right it will shoot out an error. This makes things a little bit complicated but I'll try to come up with a solution. Also, I think I need to change the testcase a little bit so that it can send the same value even in the next RND request sometimes. Thanks for pointing this out!